astropy/table/tests/test_table.py

```diff
@@ -3348,3 +3348,14 @@ def test_as_array_preserve_fill_value():
     assert tn["float"].fill_value == FLOAT_FILL
     assert tn["str"].fill_value == STR_FILL
     assert tn["cmplx"].fill_value == CMPLX_FILL
+
+
+def test_table_from_rows():
+    # see https://github.com/astropy/astropy/issues/5923
+    t1 = Table()
+    t1["a"] = [1, 2, 3]
+    t1["b"] = [2.0, 3.0, 4.0]
+
+    rows = [row for row in t1]  # noqa: C416
```

Comment:
I used a noqa comment here to keep this intention of that line very clear. Otherwise, the linter would push for
`rows = list(t1)`, which is equivalent, but a bit more opaque in my opinion.

Reply:
I could also create a list of `Rows` manually. Here I just took the example from @astrofrog for easier comparison between the test and its linked issue.